setuptools/command/build_py.py

```diff
@@ -48,14 +50,14 @@ def finalize_options(self):
             del self.__dict__['data_files']
         self.__updated_files = []
 
-    def copy_file(
+    def copy_file(  # type: ignore[override] # No overload, str support only
         self,
-        infile,
-        outfile,
-        preserve_mode=True,
-        preserve_times=True,
-        link=None,
-        level=1,
+        infile: StrPath,
+        outfile: StrPath,
+        preserve_mode: bool = True,
+        preserve_times: bool = True,
+        link: str | None = None,
+        level: object = 1,
```

Comment:
```python
        level: object = 1,
```

This one looks odd, but I suppose that is because it assumes either `int` or `boolean` values elsewhere.

Reply:
`object` can be used to mean "any parameter type" for unused params, (Typeshed has the alias `_typeshed.Unused`), but that might lead to LSP violations since subclasses should now support any `object`.

I should change it to `int` (Unless you tell me that subclasses of these classes should *not* support an "optimization level")
![image](https://github.com/user-attachments/assets/ab23b6db-86f5-4187-8003-ee3a5986c913)

Reply:
I'll need to go fix it in typeshed first (or #4689) I'll leave it "as-is" for this PR.

Reply:
Thank you @Avasam, once the CI finishes running I will squash it.